### DISCUSSION TITLE
Testing: how to hydrate atomFamily?

### DISCUSSION THREAD

**Discussion by rovolution** [pmndrs/jotai#2864] [Q&A]
Hello,

My application uses an `atomFamily` for create / caching atoms that map to a set of listings in my app (each listing has its own atom because it can be updated without affecting the others).

When writing tests for this, how do I use `useHydrateAtoms()` to hydrate the `atomFamily`? A bit confused since the `useHydrateAtoms()` hook examples seems to take a list of atoms + the mock value for the specific static atom (and atoms created via `atomFamily` can be dynamically created).

**Comment by dai-shi** [comment]
useHydrateAtoms is for some specific cases such as hydration with SSR. It's not specifically for testing, and I would avoid using it for testing and look for other solutions.

**Reply by rovolution** [reply]
why is it mentioned in the Testing documentation then? https://jotai.org/docs/guides/testing#injected-values

**Reply by dai-shi** [reply]
Good point. I think in the past it looked like a good idea. But, now looking it again, I think we should use `store` api instead for testing, if we use `<Provider>`.

**Reply by dai-shi** [reply]
```jsx
const CounterProvider = () => {
  const store = createStore()
  store.set(countAtom, 100)
  return (
    <Provider store={store}>
      <Counter />
    </Provider>
  )
}
```

**Reply by rovolution** [reply]
does `createStore()` work with `atomFamily`? or only specific atoms?

**Reply by dai-shi** [reply]
Specific atoms. You have to create atoms with your family function.

**Reply by dmaskasky** [reply]
```tsx
import {
  Provider as JotaiProvider,
  atom,
  atomFamily,
  createStore,
} from 'jotai'

const fooFamily = atomFamily(() => atom(0))

const store = createStore()

/** resets all atoms in family */
function resetFamily() {
  Array.from(fooFamily.getParams(), fooFamily)
    .forEach((fooAtom) => {
      store.set(fooAtom, fooAtom.init)
    })
}

function Component({ param }) {
  const [value, setValue] = useAtom(fooFamily(param))
  ...
}

function wrapper({ children }) {
  return (
    <JotaiProvider store={store}>
      {children}
    </JotaiProvider>
  )
}
render(Component, { wrapper, props: { param: 0 } })
...
act(resetFamily)
...

```
